backend/routes/user-routes.js

```diff
@@ -48,5 +64,121 @@ routes.post("/", async (req, res, next) => {
     }
 })
 
+routes.post("/recommendations", async (req, res, next) => {
+    try {
+        const {userId, recommendations} = req.body;
+        if (recommendations.length == 0){
+            return res.json({})
+        }
+
+        rec_posts = []
+        for (let id in recommendations){
+            rec_posts.push ({
+                "userId" : userId,
+                "postId" : parseInt(recommendations[id])
+            })
+        }
+
+        const newRecs = await prisma.recommendedPost.createMany({
+            data: rec_posts,
+            skipDuplicates: true
+        })
+
+
+        res.json(newRecs)
+    }catch (error) {
+        console.log(error)
+        res.json({"response" : "somethign went wrong"})
+    }
+})
+
+
+routes.get("/recommendations/:id", async (req, res, next) => {
+    try {
+        const recs = await prisma.recommendedPost.findMany({
+            where: {
+                userId: req.params.id
+            },
+            include : {
+                post: {
+                    include: {
+                        images: true,
+                        user: true,
+                        bids: true,
+                        likes: true
+                    }
+                }
+            }
+        })
+
+        res.json(recs)
+    }catch (error) {
+        console.log(error)
+        res.json({"response" : "something went wrong"})
+    }
+})
+
+routes.get("/liked/:id", async (req, res, next) => {
+    try {
+        const id = req.params.id
+
+        const user = await prisma.user.findUnique({
+            where: {
+                id: id
+            },
+            include : {
+                likedPosts : true
+            }
+        })
+
+        return res.json(user.likedPosts)
+
+    } catch (error) {
+        console.log(error)
+        res.json({"response" : "something went wrong"})
+    }
+})
+
+
+routes.post("/liked", async (req, res, next) => {
+    try {
+        const {userId, postId} = req.body;
+
+        const newLiked = await prisma.postLike.create({
+            data: {
+                postId,
+                userId
+            }
+        })
+        res.json(newLiked)
+    } catch (error) {
+        console.log(error)
+        res.json({"response" : "something went wrong"})
```

Comment:
we could write a util function to handle these errors and use the code

```
function handleError(error) {
    console.log(error);
    res.json({"response": "something went wrong"});
}
```

that way we have a universal function that can use for error handling across all js files -> be able to change error handling in the future if we want (maybe add a popup to the screen or something)!